src/Raven.Server/Utils/CommandLineSwitches.cs

```diff
@@ -2,28 +2,135 @@
 
 namespace Raven.Server.Utils
 {
     internal class CommandLineSwitches
     {
-        public static bool PrintServerId { get; private set; }
+        public static bool PrintServerId => ParseSwitchOption(_printIdOption);
 
-        public static bool LaunchBrowser { get; private set; }
+        public static bool LaunchBrowser => ParseSwitchOption(_browserOption);
 
-        public static bool PrintVersionAndExit { get; private set; }
+        public static bool PrintVersionAndExit => ParseSwitchOption(_versionOption);
 
-        public static string[] ParseAndRemove(string[] args)
+        public static bool RunAsService => ParseSwitchOption(_runAsServiceOption);
+
+        public static bool RegisterService => ParseSwitchOption(_registerServiceOption);
+
+        public static bool UnregisterService => ParseSwitchOption(_unregisterServiceOption);
+
+        public const string DefaultServiceName = "RavenDB";
+
+        public static string ServiceName => 
+            _serviceNameOption.Values.FirstOrDefault() ?? DefaultServiceName;
+
+        public static string CustomConfigPath =>
+            _customConfigPathOption.Values.FirstOrDefault();
+
+        public static bool ShouldShowHelp => ParseSwitchOption(_helpOption);
+
+        private static CommandLineApplication _app;
+
+        private static CommandOption _printIdOption;
+
+        private static CommandOption _browserOption;
+
+        private static CommandOption _versionOption;
+
+        private static CommandOption _registerServiceOption;
+
+        private static CommandOption _unregisterServiceOption;
+
+        private static CommandOption _runAsServiceOption;
+
+        private static CommandOption _serviceNameOption;
+
+        private static CommandOption _helpOption;
+
+        private static CommandOption _customConfigPathOption;
+
+        public static void ShowHelp()
+        {
+            _app.ShowHelp();
+        }
+
+        public static string[] Process(string[] args)
         {
             if (args == null)
                 return null;
 
-            var list = args.ToList();
-            PrintServerId = list.Remove("--print-id");
-            LaunchBrowser = list.Remove("--browser");
-            PrintVersionAndExit = list.Remove("-v") || list.Remove("--version");
-            return list.ToArray();
+            var nonConfigurationSwitches = args
+                .Where(x => Regex.IsMatch(x, @"^(--|\/)[rR]aven") == false)
+                .ToArray();
+
+            _app = new CommandLineApplication();
+            _helpOption = _app.Option(
+                "-h | -? | --help",
+                "Shows help",
+                CommandOptionType.NoValue);
+            _versionOption = _app.Option(
+                "-v | --version",
+                "Displays version and exits",
+                CommandOptionType.NoValue);
+            _printIdOption = _app.Option(
+                "--print-id", 
+                "Prints server ID upon server start", 
```

Comment:
This comes from here: https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Server/ServerWide/ServerStore.cs#L530

I introduced ability to print it from console to be used in docker scripts.